cf_xarray/parametric.py

```diff
@@ -0,0 +1,864 @@
+
+
+class AtmosphereSigma(ParamerticVerticalCoordinate):
+    """Atmosphere sigma coordinate.
+
+    Standard name: atmosphere_sigma_coordinate
+
+    Parameters
+    ----------
+    sigma : xr.DataArray
+        Vertical dimensionless coordinate.
+    ps : xr.DataArray
+        Horizontal surface pressure.
+
+    References
+    ----------
+    Please refer to the CF conventions document :
+      1. https://cfconventions.org/cf-conventions/cf-conventions.html#_atmosphere_sigma_coordinate
+    """
+
+    def __init__(self, sigma, ps, ptop):
+        self.sigma = sigma
+        self.ps = ps
+        self.ptop = ptop
+
+    def decode(self) -> xr.DataArray:
+        """Decode coordinate.
+
+        Returns
+        -------
+        xr.DataArray
+            Decoded parametric vertical coordinate.
+        """
+        p = self.ptop + self.sigma * (self.ps - self.ptop)
+
+        return p.squeeze().assign_attrs(standard_name=self.computed_standard_name)
+
+    @property
+    def computed_standard_name(self) -> str:
+        """Computes coordinate standard name."""
+        return "air_pressure"
+
+    @classmethod
+    def from_terms(cls, terms: dict):
+        """Create coordinate from terms."""
+        sigma, ps, ptop = get_terms(terms, "sigma", "ps", "ptop")
+
+        return cls(sigma, ps, ptop)
+
+
+class AtmosphereHybridSigmaPressure(ParamerticVerticalCoordinate):
+    """Atmosphere hybrid sigma pressure coordinate.
+
+    Standard name: atmosphere_hybrid_sigma_pressure_coordinate
+
+    Parameters
+    ----------
+    b : xr.DataArray
+        Component of hybrid coordinate.
+    ps : xr.DataArray
+        Horizontal surface pressure.
+    p0 : xr.DataArray
+        Reference pressure.
+    a : xr.DataArray
+        Component of hybrid coordinate.
+    ap : xr.DataArray
+        Component of hybrid coordinate.
+
+    References
+    ----------
+    Please refer to the CF conventions document :
+      1. https://cfconventions.org/cf-conventions/cf-conventions.html#_atmosphere_hybrid_sigma_pressure_coordinate
+    """
+
+    def __init__(self, b, ps, p0=None, a=None, ap=None):
+        self.b = b
+        self.ps = ps
+        self.p0 = p0
+        self.a = a
+        self.ap = ap
+
+    def decode(self) -> xr.DataArray:
+        """Decode coordinate.
+
+        Returns
+        -------
+        xr.DataArray
+            Decoded parametric vertical coordinate.
+        """
+        if self.a is None:
+            p = self.ap + self.b * self.ps
+        else:
+            p = self.a * self.p0 + self.b * self.ps
+
+        return p.squeeze().assign_attrs(standard_name=self.computed_standard_name)
+
+    @property
+    def computed_standard_name(self) -> str:
+        """Computes coordinate standard name."""
+        return "air_pressure"
+
+    @classmethod
+    def from_terms(cls, terms: dict):
+        """Create coordinate from terms."""
+        b, ps, p0, a, ap = get_terms(terms, "b", "ps", optional=("p0", "a", "ap"))
```

Comment:
if `get_terms` returned a dict with the right names then we could do 

```
cls(**get_terms(...))
```

and put all the error checking in `__post_init__` (once you move to the dataclass.

I think I prefer that.